portfolio_optimization/cuFOLIO_portfolio_optimization/CVaR/01_optimization_with_cufolio.ipynb

```
@@ -28,6 +28,41 @@
+    "        output = subprocess.check_output(\"nvidia-smi\", shell=True).decode()\n",
+    "        display(HTML(f\"\"\"\n",
+    "        <div style=\"border:2px solid #4CAF50;padding:10px;border-radius:10px;background:#e8f5e9;\">\n",
+    "            <h3>✅ GPU is enabled</h3>\n",
+    "            <pre>{output.splitlines()[2]}</pre>\n",
+    "        </div>\n",
+    "        \"\"\"))\n",
+    "    except Exception:\n",
+    "        display(HTML(\"\"\"\n",
+    "        <div style=\"border:2px solid red;padding:15px;border-radius:10px;background:#ffeeee;\">\n",
+    "            <h3>⚠️ GPU not detected!</h3>\n",
+    "            <p>This notebook requires a <b>GPU runtime</b>.</p>\n",
```

Ruff (0.14.0)

`from cuopt.linear_programming.solver.solver_parameters import *` used; unable to detect undefined names

(F403)